From 35892d959c0c8bb49df6454a5be6ad334325eac8 Mon Sep 17 00:00:00 2001
From: Karthikeyan Kathirvel <quic_kathirve@quicinc.com>
Date: Fri, 13 May 2022 17:22:58 +0530
Subject: [PATCH] ath11k: add hw_param to control the target mem mode

Adding hw_param to set the target mem mode across different chipsets

Signed-off-by: Karthikeyan Kathirvel <quic_kathirve@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/core.c | 4 ++++
 drivers/net/wireless/ath/ath11k/hw.h   | 1 +
 drivers/net/wireless/ath/ath11k/qmi.c  | 2 +-
 3 files changed, 6 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/ath/ath11k/core.c
+++ b/drivers/net/wireless/ath/ath11k/core.c
@@ -124,6 +124,7 @@ static const struct ath11k_hw_params ath
 		.supports_shadow_regs = false,
 		.idle_ps = false,
 		.cold_boot_calib = true,
+		.fw_mem_mode = 0,
 		.supports_suspend = false,
 		.hal_desc_sz = sizeof(struct hal_rx_desc_ipq8074),
 		.reo_dest_ring_map_shift = HAL_REO_DEST_RING_CTRL_HASH_RING_SHIFT,
@@ -185,6 +186,7 @@ static const struct ath11k_hw_params ath
 		.supports_shadow_regs = false,
 		.idle_ps = false,
 		.cold_boot_calib = true,
+		.fw_mem_mode = 0,
 		.supports_suspend = false,
 		.hal_desc_sz = sizeof(struct hal_rx_desc_ipq8074),
 		.reo_dest_ring_map_shift = HAL_REO_DEST_RING_CTRL_HASH_RING_SHIFT,
@@ -234,6 +236,7 @@ static const struct ath11k_hw_params ath
 		.supports_shadow_regs = true,
 		.idle_ps = true,
 		.cold_boot_calib = false,
+		.fw_mem_mode = 0,
 		.supports_suspend = true,
 		.hal_desc_sz = sizeof(struct hal_rx_desc_ipq8074),
 		.reo_dest_ring_map_shift = HAL_REO_DEST_RING_CTRL_HASH_RING_SHIFT,
@@ -283,6 +286,7 @@ static const struct ath11k_hw_params ath
 		.supports_shadow_regs = false,
 		.idle_ps = false,
 		.cold_boot_calib = true,
+		.fw_mem_mode = 2,
 		.supports_suspend = false,
 		.hal_desc_sz = sizeof(struct hal_rx_desc_qcn9074),
 		.spectral_fft_pad_sz = 2,
@@ -362,6 +366,7 @@ static const struct ath11k_hw_params ath
 		.ce_fwlog_enable = true,
 		.fwmem_mode_change = false,
 		.cold_boot_calib = true,
+		.fw_mem_mode = 1,
 		.is_qdss_support = false,
 		.max_tx_ring = 1,
 		.wakeup_mhi = false,
@@ -428,6 +433,7 @@ static const struct ath11k_hw_params ath
 		.reo_dest_ring_map_shift = HAL_IPQ5018_REO_DEST_RING_CTRL_HASH_RING_SHIFT,
 		.ce_fwlog_enable = true,
 		.cold_boot_calib = true,
+		.fw_mem_mode = 1,
  		.fwmem_mode_change = false,
  		.is_qdss_support = true,
 		.max_tx_ring = 1,
--- a/drivers/net/wireless/ath/ath11k/hw.h
+++ b/drivers/net/wireless/ath/ath11k/hw.h
@@ -201,6 +201,7 @@ struct ath11k_hw_params {
 	bool supports_shadow_regs;
 	bool idle_ps;
 	bool cold_boot_calib;
+	int fw_mem_mode;
 	bool supports_suspend;
 	u32 hal_desc_sz;
 	u32 m3_addr;
--- a/drivers/net/wireless/ath/ath11k/qmi.c
+++ b/drivers/net/wireless/ath/ath11k/qmi.c
@@ -4655,7 +4655,7 @@ int ath11k_qmi_init_service(struct ath11
 	memset(&ab->qmi.target_mem, 0, sizeof(struct target_mem_chunk));
 	ab->qmi.ab = ab;
 
-	ab->qmi.target_mem_mode = ATH11K_QMI_TARGET_MEM_MODE;
+	ab->qmi.target_mem_mode = ab->hw_params.fw_mem_mode;
 
 	if (ab->hw_params.fwmem_mode_change) {
 		ab->qmi.target_mem_mode = fwmem_mode;
